### PR TITLE
Prometheus: Limit max number of metrics to be passed into the querybuilder virtualized select

### PR DESCRIPTION
**Pull request by gtk-grafana**
Limit max number of metrics to be passed into the querybuilder virtualized select, and display warning when limit is exceeded.

As short-term resolution for https://github.com/grafana/support-escalations/issues/4008

**What this PR does / why we need it**:
If too many metrics are saved to the react state the application runs out of memory and crashes, since some customers are trying to feed over 10k metrics (potentially hundreds of megabytes of raw data) we need to keep the UI from crashing.

**Which issue(s) this PR fixes**:
https://github.com/grafana/support-escalations/issues/4008

**Special notes for your reviewer**:
This is not an ideal solution, but hopefully this allows users with huge amount of metrics to be able to filter things down enough to make use of the query builder.

We'll want to address the query-builder's inability to handle more then 10k records in another ticket.

**Summary:**
This PR simply sets a max limit (10k to match the native prometheus UI) of results we save to the react state, which will prevent crashes from huge results eating up all the memory, but I’m worried about unintended consequences of this change.

The good: When you’re trying to return a million metrics, you can still use the querybuilder! After you filter things down by selecting a few label filters, you typically get a response of less than 10k metrics, and users can still build queries without knowing promql. I would argue that this is better than crashing the application.

The bad: If you try to search for stuff, it might not return a response if it wasn’t in the first 10k results (see screenshots below). You can work around this by adding filters as long as that returns a response of less than 10k. I could see an argument that this UX is not acceptable to release, and we should wait to release a better solution, or that users should be able to configure their own limits instead of imposing an arbitrary number of metrics.
image
image

TL;DR: Getting the query builder to work with more than 10k results is going to require some significant changes to the UI/UX, and the backend in order to support this properly. Galen created this [PR](https://github.com/grafana/grafana/pull/56897/files) as a quick-fix, but is wary about releasing a sub-par UX, which could confuse users or cause other trouble.

### CODE REVIEW

**Comment by gtk-grafana** [comment]
Looks like I need to do a bit more investigation on this, the setState isn't responsible for the out of memory error, so this solution is wrong.
